src/alpaka/AlpakaCore/HistoContainer.h

```diff
@@ -0,0 +1,354 @@
+        const uint32_t gridDimension(alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Elems>(acc)[0u]);
+        const auto &[firstElementIdxNoStride, endElementIdxNoStride] =
+            cms::alpakatools::element_global_index_range_truncated(acc, Vec1::all(nt));
+        for (uint32_t threadIdx = firstElementIdxNoStride[0u], endElementIdx = endElementIdxNoStride[0u];
+             threadIdx < nt;
+             threadIdx += gridDimension, endElementIdx += gridDimension) {
+          for (uint32_t i = threadIdx; i < std::min(endElementIdx, nt); ++i) {
+            auto off = alpaka_std::upper_bound(offsets, offsets + nh + 1, i);
+            assert((*off) > 0);
+            int32_t ih = off - offsets - 1;
+            assert(ih >= 0);
+            assert(ih < int(nh));
+            h->count(acc, v[i], ih);
+          }
+        }
+      }
+    };
+
+    struct fillFromVector {
+      template <typename T_Acc, typename Histo, typename T>
+      ALPAKA_FN_ACC void operator()(const T_Acc &acc,
+                                    Histo *__restrict__ h,
+                                    uint32_t nh,
+                                    T const *__restrict__ v,
+                                    uint32_t const *__restrict__ offsets) const {
+        const uint32_t nt = offsets[nh];
+        const uint32_t gridDimension(alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Elems>(acc)[0u]);
+        const auto &[firstElementIdxNoStride, endElementIdxNoStride] =
+            cms::alpakatools::element_global_index_range_truncated(acc, Vec1::all(nt));
+
+        for (uint32_t threadIdx = firstElementIdxNoStride[0u], endElementIdx = endElementIdxNoStride[0u];
+             threadIdx < nt;
+             threadIdx += gridDimension, endElementIdx += gridDimension) {
+          for (uint32_t i = threadIdx; i < std::min(endElementIdx, nt); ++i) {
+            auto off = alpaka_std::upper_bound(offsets, offsets + nh + 1, i);
+            assert((*off) > 0);
+            int32_t ih = off - offsets - 1;
+            assert(ih >= 0);
+            assert(ih < int(nh));
+            h->fill(acc, v[i], i, ih);
+          }
+        }
+      }
+    };
+
+    struct launchZero {
+      template <typename T_Acc, typename Histo>
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE __attribute__((always_inline)) void operator()(const T_Acc &acc,
+                                                                                    Histo *__restrict__ h) const {
+        const auto &[firstElementIdxGlobal, endElementIdxGlobal] =
+            cms::alpakatools::element_global_index_range_truncated(acc, Vec1::all(Histo::totbins()));
+
+        for (uint32_t i = firstElementIdxGlobal[0u]; i < endElementIdxGlobal[0u]; ++i) {
+          h->off[i] = 0;
+        }
+      }
+    };
+
+    template <typename Histo>
+    ALPAKA_FN_HOST ALPAKA_FN_INLINE __attribute__((always_inline)) void launchFinalize(
+        Histo *__restrict__ h,
+        const ALPAKA_ACCELERATOR_NAMESPACE::DevAcc1 &device,
```

Comment:
After removing the memory allocation, the `device` parameter is not needed anymore
```suggestion
```